Microsoft.Toolkit.Uwp.UI.Controls/TokenizingTextBox/TokenizingTextBox.cs

```diff
@@ -0,0 +1,228 @@
+        {
+            QueryTextChanged?.Invoke(sender, args);
+        }
+
+        private void TokenizingTextBox_ClearClicked(TokenizingTextBoxItem sender, RoutedEventArgs args)
+        {
+            RemoveToken(sender);
+        }
+
+        private void TokenizingTextBox_Click(object sender, RoutedEventArgs e)
+        {
+            if (sender is TokenizingTextBoxItem item)
+            {
+                if (!item.IsSelected && !CoreWindow.GetForCurrentThread().GetKeyState(VirtualKey.Control).HasFlag(CoreVirtualKeyStates.Down))
+                {
+                    foreach (var child in _wrapPanel.Children)
+                    {
+                        if (child is TokenizingTextBoxItem childItem)
+                        {
+                            childItem.IsSelected = false;
+                        }
+                    }
+                }
+
+                item.IsSelected = !item.IsSelected;
+
+                var clickedItem = item.Content;
+                TokenItemClicked?.Invoke(this, clickedItem);
+            }
+        }
+
+        private void AddToken(object data)
+        {
+            var item = new TokenizingTextBoxItem()
+            {
+                Content = data,
+                ContentTemplateSelector = TokenItemTemplateSelector,
+                ContentTemplate = TokenItemTemplate,
+                Style = TokenItemStyle
+            };
+            item.Click += TokenizingTextBox_Click;
+            item.ClearClicked += TokenizingTextBox_ClearClicked;
+
+            var removeMenuItem = new MenuFlyoutItem
+            {
+                Text = "Remove"
+            };
+            removeMenuItem.Click += (s, e) => RemoveToken(item);
+            var menuFlyout = new MenuFlyout();
+            menuFlyout.Items.Add(removeMenuItem);
+            item.ContextFlyout = menuFlyout;
+
+            var i = _wrapPanel.Children.Count - 1;
+            _wrapPanel.Children.Insert(i, item);
+
+            TokenItemAdded?.Invoke(this, data);
+        }
+
+        private void RemoveToken(TokenizingTextBoxItem item)
+        {
+            var itemIndex = Math.Max(_wrapPanel.Children.IndexOf(item) - 1, 0);
+            _wrapPanel.Children.Remove(item);
+
+            if (_wrapPanel.Children[itemIndex] is Control control)
+            {
+                control.Focus(FocusState.Programmatic);
+            }
+        }
+
+        private void SelectAll()
+        {
+            foreach (var child in _wrapPanel.Children)
+            {
+                if (child is TokenizingTextBoxItem item)
+                {
+                    item.IsSelected = true;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the string representation of each token item, concatenated and delimeted by the <see cref="TokenDelimiter"/>
+        /// </summary>
+        /// <returns>Untokenized text string</returns>
+        public string GetUntokenizedText()
```

Comment:
Should you be able to pass in an optional alternate delimiter?  thinking in the default case there's no delimiter so all the text would be smooshed together...  if none is provided it'd use the property, otherwise it'd use the one provided?

Reply:
Good idea. I actually think it's a good idea to stop using the TokenDelimiter property altogether, and instead go with:

`public string GetUntokenizedText(string tokenDelimiter = " ,")
        { ...`

The TokenDelimiter property is more of an input mechanic.